domain/modeldefaults/state/state.go

```diff
@@ -529,25 +681,30 @@ WHERE key IN ($attrs[:])
 AND region_uuid = $cloudRegion.uuid;
 `, toRemove, region)
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Capture(err)
 	}
 
-	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		err := tx.Query(ctx, selectStmt, cld, region).Get(&region)
 		if errors.Is(err, sqlair.ErrNoRows) {
-			return interrors.Errorf("cloud %q region %q %w", cloudUUID, regionName, errors.NotFound)
+			return errors.Errorf(
```

Comment:
Not sure why the extra error annotation "removing cloud %q region defaults" was moved inside the txn closure and lines 544-552 deleted. Our best practice is to just add enough context for any errors running the txn operations and the db.Txn() caller then adds the overall context for the state operation being performed.

Reply:
The error messages are fundamentally different based on the conditions. The conditions can only be asserted within the tx. There is no other way to perform this if we can't interrogate the error outside of the tx.